### ISSUE TITLE
SWA Callback triggers OOM

### ISSUE THREAD

**Issue by tchaton** [PyTorchLightning/pytorch-lightning#9480]
I was using the previous version of lightning and SWA worked fine. However, when I updated to the latest version of lightning, the batchnorm forward pass run exceeds my GPU memory cap. I suspect this is due to gradient accumulation (I did not use any) in the source code. Can anyone offer any suggestion to address this, I can't  revert to prev version due to dependencies.

_Originally posted by @HarryLe95 in https://github.com/PyTorchLightning/pytorch-lightning/discussions/9469_

**Comment by HarryLe95** [comment]
The previous version was pytorch-lightning 1.2.4. The updated one is 1.4.5. I have tried downgrading and upgrading lightning to replicate the problem. On normal training, the memory consumption was 12GB, and at the last batchnorm run was slightly above 12GB with 1.2.4 lightning, and 17GB with 1.4.5 lightning. The SWA callback code seems fine - I have taken the source code from 1.2.4 and plugged into 1.4.5 and the problem still persists, so I suspect the difference was due to the loop class or something similar. I hope this helps identifying the problem.
